### ISSUE TITLE
Add support for "multiple" select

### ISSUE THREAD

**Issue by cedricmessiant** [mozilla-services/react-jsonschema-form#41]
I need to be able to create select fields with the possibilty to select multiple values.
I can work on it but I'm not sure which way I should go with jsonschema.

My intuition is to use a schema that looks like this : 

>```
myField: {
  type: 'array',
  title: 'My field',
  items: {
    type: 'string',
    enum: ['foo', 'bar'],
  }
}
```

maybe with an alternative widget `multiselect` for example.

What are your opinions about this ?

I will probably also need to support enumNames (See https://github.com/json-schema/json-schema/wiki/enumNames-(v5-proposal) )

**Comment by almet** [comment]
I'll let @n1k0 ponder, he's our local json-schema expert (if you ask him, he'll say he's not, but that's lies)

**Comment by n1k0** [comment]
@almet Haha, hmpf, thanks I guess.

Yes, we need to deal with a specific kind of array widget when the schema is constrained with an enum of strings (possibly numbers, integers, datetimes too) so we can render a select field (or a list of checkbox btw) instead of the regular `ArrayField`.

The only thing is that with such a schema, one might want to allow multiple occurrences of an allowed `enum` value in the resulting array, eg. `["foo", "bar", "foo"]`. Reading this [thread from 2009](https://groups.google.com/forum/#!topic/json-schema/NCNWSfnE2QA) about just this is slightly worrying though, as I couldn't find any followup to this specific problem. Hope it's just me who can't search for things.

Also, re `enumNames` support, there's #25 issue filed already.

**Comment by n1k0** [comment]
Actually why googling for solutions I've just stumbled upon the [Choices v5 proposal](https://github.com/json-schema/json-schema/wiki/choices-%28v5-proposal-to-enhance-enum%29), which is probably something we'll want to support it makes it to the spec eventually.

**Comment by n1k0** [comment]
And so for ensuring a multiple choices field can't be mistaken for a multiple occurrences of choices field, there's is that [UniqueItems](https://github.com/json-schema/json-schema/wiki/Uniqueitems) proposal.

**Comment by cedricmessiant** [comment]
Okay, so to sum up I can start working on supporting schemas like this : 

>```
myField: {
  type: 'array',
  title: 'My field',
  items: {
    enum: [
      'foo',
      'bar',
    ],
    type: 'string'
  },
  uniqueItems: true,
}
```

resulting on a multiple select (and maybe alternatively a list of checkboxes). Is that it ?

**Comment by n1k0** [comment]
> Is that it ?

That's a good summary of it, yes. Thanks you so much on planning to help and contribute!

**Comment by cedricmessiant** [comment]
I just finished the work on this issue. However, I can not push my branch upstream to make my pull request. Could you give me the rigths on the repo or do you prefer that I make my pull request from a fork ?

**Comment by Natim** [comment]
Yes please do it from a fork, you could also give us some contributors roles on your fork if you want us to collaborate on your branch.

**Comment by n1k0** [comment]
Fixed by #42.
